securedrop/tests/test_encryption.py

```diff
@@ -0,0 +1,355 @@
+        file_to_encrypt_path = Path(__file__)
+        with file_to_encrypt_path.open() as file_to_encrypt:
+
+            # When the source tries to encrypt the file
+            # It succeeds
+            encrypted_file_path = tmp_path / "file.gpg"
+            encryption_mgr.encrypt_source_file(
+                file_in=file_to_encrypt,
+                encrypted_file_path_out=encrypted_file_path,
+            )
+
+            # And the output file contains the encrypted data
+            encrypted_file = encrypted_file_path.read_bytes()
+            assert encrypted_file
+
+        # And the journalist is able to decrypt the file
+        with import_journalist_private_key(encryption_mgr):
+            decrypted_file = encryption_mgr._gpg.decrypt(encrypted_file).data
+        assert decrypted_file.decode() == file_to_encrypt_path.read_text()
+
+        # And the source or anyone else is NOT able to decrypt the file
+        # For GPG 2.1+, a non-null passphrase _must_ be passed to decrypt()
+        assert not encryption_mgr._gpg.decrypt(encrypted_file, passphrase="test 123").ok
+
+    def test_encrypt_and_decrypt_journalist_reply(self, source_app, test_source, tmp_path):
+        # Given a source user with a key pair in the default encryption manager
+        source_user1 = test_source["source_user"]
+        encryption_mgr = EncryptionManager.get_default()
+
+        # And another source with a key pair in the default encryption manager
+        with source_app.app_context():
+            source_user2 = create_source_user(
+                db_session=db.session,
+                source_passphrase=PassphraseGenerator.get_default().generate_passphrase(),
+                source_app_storage=source_app.storage,
+            )
+        encryption_mgr.generate_source_key_pair(source_user2)
+
+        # When the journalist tries to encrypt a reply to source1
+        # It succeeds
+        journalist_reply = "s3cr3t message"
+        encrypted_reply_path = tmp_path / "reply.gpg"
+        encryption_mgr.encrypt_journalist_reply(
+            for_source_with_filesystem_id=source_user1.filesystem_id,
+            reply_in=journalist_reply,
+            encrypted_reply_path_out=encrypted_reply_path,
+        )
+
+        # And the output file contains the encrypted data
+        encrypted_reply = encrypted_reply_path.read_bytes()
+        assert encrypted_reply
+
+        # And source1 is able to decrypt the reply
+        decrypted_reply = encryption_mgr.decrypt_journalist_reply(
+            for_source_user=source_user1, ciphertext_in=encrypted_reply
+        )
+        assert decrypted_reply
+        assert decrypted_reply == journalist_reply
+
+        # And source2 is NOT able to decrypt the reply
+        with pytest.raises(GpgDecryptError):
+            encryption_mgr.decrypt_journalist_reply(
+                for_source_user=source_user2, ciphertext_in=encrypted_reply
+            )
```

Comment:
Without `default-cache-ttl 0`, this part of the test would fail: source2 would be able to decrypt the reply.